packages/rtk-query-graphql-request-base-query/src/GraphqlBaseQueryTypes.ts

```diff
@@ -0,0 +1,50 @@
+import type { GraphQLClient } from 'graphql-request'
+import type { ThunkDispatch } from 'redux-thunk'
+
+type P = Parameters<GraphQLClient['request']>
+export type Document = P[0]
+export type Variables = P[1]
+export type RequestHeaders = P[2]
+
+export interface GraphqlBaseQueryApi {
+  dispatch: ThunkDispatch<any, any, any>
+  getState: () => unknown
+}
+
+export type GraphqlRequestBaseQueryArgs = {
+  options:
+    | {
+        url: string
+        requestHeaders?: RequestHeaders
+      }
+    | { client: GraphQLClient }
+  prepareHeaders?: (
+    headers: Headers,
+    api: Pick<GraphqlBaseQueryApi, 'getState'>
+  ) => MaybePromise<Headers>
+}
+
+export type BaseQueryFn<
+  Args = any,
+  Result = unknown,
```

Comment:
I'm not sure if making Result in any way configurable other than `unknown` makes any sense here - that would have to be a type of all possible results of all possible queries to send to the server. Do you have an example on how you would use that?

Reply:
Hmm, not quite sure on that one as well. Actually, `BaseQueryFn` has (almost) the same signature as https://github.com/reduxjs/redux-toolkit/blob/master/packages/toolkit/src/query/baseQueryTypes.ts#L34. I added that new type with almost the same signature to be able to change the type definition for `api` param to the newly-added `GraphqlBaseQueryApi` interface:

````
export type BaseQueryFn<Args = any, Result = unknown, Error = unknown, DefinitionExtraOptions = {}, Meta = {}> = (
  args: Args,
  api: GraphqlBaseQueryApi,
  extraOptions: DefinitionExtraOptions,
) => MaybePromise<QueryReturnValue<Result, Error, Meta>>;
````

Reply:
Hey, @phryneas -- hope you're doing alright.

Checking back on this one, do you have any other question/suggestion? Something I should be looking into?

Thanks!

Reply:
Sorry, I've got quite turbulent times right now - including a Covid infection. I'll try to get back to things next work, but no guarantees at the moment.

Reply:
Actually, now that I could take a closer look at it, that type should not be copied over at all. I'll push another commit in a few minutes, can you please check if that works as expected for you?

Reply:
Yes, sure! I can test it out right away!

Reply:
Yes! That worked. I just had to pass in `requestHeaders` in `new Headers(stripUndefined({))` and it's all working great.